### ISSUE TITLE
Convention for cuda header files

### ISSUE THREAD

**Issue by Rombur** [dealii/dealii#5727]
Working on cuda, I realized that I don't have a real convention for the cuda header files (some end with `.h` and others with `.cuh`). I wanted to use `.cuh` if the header contained some cuda code and `.h` if it didn't but then what happens if we add something to the interface, we can't really rename the files. I think the most common convention is that header files corresponding to a `.cu`  use  `.cuh`. We could do that or we could use `.h` everywhere. We already have cuda code that must be in `.templates.h` (for example in `read_write_vector.templates.h`), so we will always have cuda code in `.h` file. Note that this is just a convention, `nvcc` doesn't care about the file extension of the header files. What do you guys think? Whatever we decide, I need to rename the files before the release because right now it is a mess :disappointed:

**Comment by kronbichler** [comment]
Since `nvcc` does not care, I would go with `.h`. When I started using the `.cuh` files of Karl Ljungkvist I was also confused and looked for a deeper meaning. So I guess we would be consistent if we simply used the same convention. If I remember correctly, `cmake` will use different colors for the `cc` files (or `cu` files)  once control is headed to `nvcc` (blue instead of green), but that should not be affected I think.
